### DISCUSSION TITLE
Nornir - How to pass the secret password for Napalm

### DISCUSSION THREAD

**Discussion by gaffar10** [nornir-automation/nornir#720] [General]
Can someone please help to provide me the syntax to pass the optional_args, secret?

For example, with input, get. pass, Env variable, GPG encryption method, I was able to pass the username and password successfully.

username = input("Enter Username: ")
password = input("Enter Password: ")   OR password = getpass.getpass()
nr.inventory.defaults.username = username
nr.inventory.defaults.password = password

However, I don't know the syntax to pass the secret password. I tried with the below options but it doesn't work.

secret = input("Enter Enable Password: ")  OR  secret = getpass.getpass()

nr.inventory.defaults.secret = secret
nr.inventory.defaults.connection_options.napalm.extras.optional_args.secret = secret
nr.inventory.defaults.connection_options['napalm'] = ConnectionOptions(extras={"optional_args":{"secret":"secret"}})

Thanks in advance

**Comment by ktbyers** [comment]
I answered this same question from you here:

https://github.com/nornir-automation/nornir_napalm/issues/17


Please only ask a given question in one place.
